src/cheri_insts.sail

```diff
@@ -2021,7 +2033,7 @@ function handle_store_cond_cap_via_cap(cs2, auth_idx, auth_val, vaddrBits) = {
     RETIRE_FAIL
   } else if match_reservation(vaddrBits) == false then {
     /* cannot happen in rmem */
-    C(cs2) = int_to_cap(EXTZ(0b1));
+    X(rd) = EXTZ(0b1);
```

Comment:
Hmm, for explicit SC of a cap this was deliberately written using `C` so it works for split register files

Reply:
That's a good point, I didn't think about that. Since it only ever writes an integer I thought it made sense to use the integer register file. In a split register file world the versions of the instructions that have a source / dest operand could read from a capability register and write to an integer register with the same number. That would avoid having to do a `cgetaddr` in order to branch on the return value but I guess it would be a nightmare for the compiler's register allocator?

Reply:
I've now pushed a version where the destination register is either a capability or integer depending on the instruction.